External/Plugins/ASCompletion/Completion/ASGenerator.cs

```diff
@@ -2035,9 +2037,21 @@ public static int GetEndOfStatement(int startPos, int endPos, ScintillaControl s
                 {
                     statementEnd++;
                     continue;
+                }
+                char c = (char)sci.CharAt(statementEnd++);
+                if (c == '\"' && sQuotes == 0)
+                {
+                    if (statementEnd <= 1 || (char)sci.CharAt(statementEnd - 2) == '\\') continue;
+                    if (dQuotes == 0) dQuotes++;
+                    else dQuotes--;
+                }
+                else if (c == '\'' && dQuotes == 0)
+                {
+                    if (statementEnd <= 1 || (char)sci.CharAt(statementEnd - 2) == '\\') continue;
+                    if (sQuotes == 0) sQuotes++;
+                    else sQuotes--;
```

Comment:
This function already uses a Scintilla instance to know about comments why not doing the same for strings? I see a problem with it, also existing in the version without it... multiline strings. It's not handled properly here, and the bundled Scintilla doesn't support them either, but then it's delegating it to when a more proper Scintilla is added. Also, if `startPos` would happen to be in the middle of a string this wouldn't work properly.

Reply:
Thanks. I have to check this out.